### ISSUE TITLE
Use the RemoteAPI but the docs is incomplete

### ISSUE THREAD

**Issue by tracylihui** [docker/docker#20536]
Description of problem:
My docker version is 1.9.1. I use the RemoteAPI(https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.21.md) to have some tests. And I found that the docs is incomplete. For example, 'GET /containers/(id)/json' but 'GET /containers/(name)/json' is right too. Maybe we should complete this.
@thaJeztah @moxiegirl

**Comment by tracylihui** [comment]
I have read the docs(https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.21.md), there is no mention about 'GET /containers/(name)/json' .How do you think? @thaJeztah @moxiegirl

**Comment by ozlerhakan** [comment]
hi @tracylihui , if you scroll down a bit you will see some examples of other rest queries including images and containers whose `id `can refer to id, first 12-char of id, or name. 

you can try by yourself for examples: 
```
//for a container:
POST /containers/oldname/rename?name=newname

//images:
get /images/3876b81b5a81/json  
get /images/ubuntu:14.04/json
```

**Comment by thaJeztah** [comment]
I think it would be good to include that in the documentation, so suggestions/PR welcome

**Comment by tracylihui** [comment]
@ozlerhakan  I find what say, but I mean that it would be good to include name, such as 'GET /containers/(id or name)/json'. How do you think? It's OK? @thaJeztah

**Comment by tracylihui** [comment]
#dibs

**Comment by thaJeztah** [comment]
I think that should work. If you want to work on that, can you also update the older versions of the documentation? Otherwise people may think that this is something that's changed over time (which, afaik, isn't the case)

**Comment by tracylihui** [comment]
OK. I will hava a PR about docker version 1.9.x and docker_remote_api_v1.21 . I hava tests about this . Then my classmates and i will test about older versions of the documentation.

**Comment by tracylihui** [comment]
@hsinko @longliqiang88

**Comment by longliqiang88** [comment]
OK

**Comment by hsinko** [comment]
@tracylihui  OK.

**Comment by ozlerhakan** [comment]
you're right @tracylihui

**Comment by LK4D4** [comment]
Looks like was fixed in #20557
